src/main/scala/com/cognite/sdk/scala/v1/resources/dataModels.scala

```diff
@@ -60,28 +56,137 @@ class DataModels[F[_]](val requestSession: RequestSession[F])
 }
 
 object DataModels {
-  implicit val dataModelPropertyIndexEncoder: Encoder[DataModelPropertyIndex] =
-    deriveEncoder[DataModelPropertyIndex]
+  implicit val dataModelIdentifierEncoder: Encoder[DataModelIdentifier] =
+    Encoder
+      .encodeIterable[String, Seq]
+      .contramap[DataModelIdentifier] {
+        case DataModelIdentifier(Some(space), model) =>
+          Seq(space, model)
+        case DataModelIdentifier(_, model) =>
+          Seq(model)
+      }
+
+  implicit val bTreeIndexEncoder: Encoder[BTreeIndex] =
+    deriveEncoder[BTreeIndex]
+  implicit val dataModelPropertyTypeEncoder: Encoder[PropertyType] =
+    Encoder.encodeString.contramap[PropertyType](_.code)
+  implicit val dataModelPropertyIndexesEncoder: Encoder[DataModelIndexes] =
+    deriveEncoder[DataModelIndexes]
   implicit val dataModelPropertyEncoder: Encoder[DataModelProperty] =
     deriveEncoder[DataModelProperty]
-  implicit val dataModelEncoder: Encoder[DataModel] = deriveEncoder[DataModel]
-  implicit val dataModelItemsEncoder: Encoder[Items[DataModel]] = deriveEncoder[Items[DataModel]]
+  implicit val uniquenessConstraintEncoder: Encoder[UniquenessConstraint] =
+    deriveEncoder[UniquenessConstraint].mapJson(init =>
+      init.hcursor
+        .downField("uniqueProperties")
+        .set(
+          Json.fromValues(
+            init.hcursor
+              .downField("uniqueProperties")
+              .values
+              .map(_.map(v => Json.fromFields(Seq("property" -> v))))
+              .getOrElse(throw new Exception("uniqueProperties is required"))
+          )
+        )
+        .top
+        .getOrElse(throw new Exception("uniqueProperties is required"))
+    )
```

Comment:
I think we're actually better off with [defining the Encoder explicitly](https://circe.github.io/circe/codecs/custom-codecs.html) here. Note that this can be condensed even more, but I think it's better to keep it like this for clarity (the `new Encoder[..` actually isn't strictly needed here).
```
  implicit val uniquenessConstraintEncoder: Encoder[UniquenessConstraint] =
    new Encoder[UniquenessConstraint] {
      final def apply(uc: UniquenessConstraint): Json = Json.fromValues(uc.uniqueProperties.map(Json.fromString))
    }
```
Also not sure we need the validation that `uniqueProperties` is present, isn't that something we can enforce from Scala instead? If we're trying to encode an instance of `UniquenessConstraint`, that someone was able to create it in the first place guarantees that `uc` is defined.. if you're looking to validate it's not the empty Seq, you can change the type to a non-nullable-seq.